lncfg/remotesigner.go

```diff
@@ -6,39 +6,168 @@ import (
 )
 
 const (
-	// DefaultRemoteSignerRPCTimeout is the default timeout that is used
-	// when forwarding a request to the remote signer through RPC.
+	// DefaultRemoteSignerRPCTimeout is the default connection timeout
+	// that is used when connecting to the remote signer or watch-only node
+	// through RPC.
 	DefaultRemoteSignerRPCTimeout = 5 * time.Second
+
+	// DefaultRequestTimeout is the default timeout used for requests to and
+	// from the remote signer.
+	DefaultRequestTimeout = 5 * time.Second
+
+	// DefaultStartupTimeout is the default startup timeout used when a
+	// watch-only node with 'remotesigner.allowinboundconnection' set to
+	// true waits for the remote signer to connect.
+	DefaultStartupTimeout = 5 * time.Minute
 )
 
-// RemoteSigner holds the configuration options for a remote RPC signer.
+// RemoteSigner holds the configuration options for how to connect to a remote
+// signer. Only a watch-only node specifies this config.
 //
 //nolint:ll
 type RemoteSigner struct {
-	Enable           bool          `long:"enable" description:"Use a remote signer for signing any on-chain related transactions or messages. Only recommended if local wallet is initialized as watch-only. Remote signer must use the same seed/root key as the local watch-only wallet but must have private keys."`
-	RPCHost          string        `long:"rpchost" description:"The remote signer's RPC host:port"`
-	MacaroonPath     string        `long:"macaroonpath" description:"The macaroon to use for authenticating with the remote signer"`
-	TLSCertPath      string        `long:"tlscertpath" description:"The TLS certificate to use for establishing the remote signer's identity"`
-	Timeout          time.Duration `long:"timeout" description:"The timeout for connecting to and signing requests with the remote signer. Valid time units are {s, m, h}."`
-	MigrateWatchOnly bool          `long:"migrate-wallet-to-watch-only" description:"If a wallet with private key material already exists, migrate it into a watch-only wallet on first startup. WARNING: This cannot be undone! Make sure you have backed up your seed before you use this flag! All private keys will be purged from the wallet after first unlock with this flag!"`
+	// Enable signals if this node is a watch-only node in a remote signer
+	// setup.
+	Enable bool `long:"enable" description:"Use a remote signer for signing any on-chain related transactions or messages. Only recommended if local wallet is initialized as watch-only. Remote signer must use the same seed/root key as the local watch-only wallet but must have private keys."`
+
+	// AllowInboundConnection is true if the signer node will connect to this node.
+	AllowInboundConnection bool `long:"allowinboundconnection" description:"Signals that we allow an inbound connection from a remote signer to this node."`
```

Comment:
I actually do think explicitly requiring this flag/boolean to be set is a better UX than not deriving that from checking the `ConnectionCfg` values because:
1. It makes it very clear and explicit to users what they are toggling.
2. I think we should phase out the default behaviour that the remote signer allows an inbound connection over time as well, and also explicitly require that the user specifies a corresponding flag on In `watchonlynode` config on the remote signer, to make that security aspect clearer to the user. I suggest introducing that flag for the `watchonlynode` config in https://github.com/lightningnetwork/lnd/pull/9293, and then later require that it's set.

I therefore didn't address this feedback. If you still think that doesn't motivate the existence of this flag, let me know!

Ps. I'm open to other naming suggestions for the flag :).